### PR TITLE
Introducing inverse

### LINKED ISSUE #35: Parse "INVERSE" attributes from IFC schema
Many (previously existing) functionalities can only be implemented, when the INVERSE attributes in the IFC schema are parsed and corresponding members generated. One of such functionalities is #34 .

### PR DESCRIPTION
**Pull request by pjanck**
First, #195 should be merged. This Pull Request builds on top - all changes with paths starting with `UnitTests` can be safely ignored.

Fixes #35 .

**tl;dr:**

INVERSE parameters are now parsed and filled with data when reading.

**What has been done:**
- `Entity` has a new function `setInverseCounterpart` that links this->attribute with the other->attribute
- `EntityAttribute` now knows about its inverse status, as well as about the inverse's counterpart's name (only the true inverse has its `inverse` flag set to true)
- `Schema::linkInverses()` links all inverse attributes among eachother
- parser now understands the inverse statements
- generator was adapted accordingly:
  - each entity now has a function `void linkInverse()` that links itself to the inverse attribute of all attributes with inverse counterparts (`this->attr->inverseAttribute->add(this)`
  - `main` links the inverses just before generating the files

**How to test?**

Using the UT_schema_v3:
- both `IfcAlignment2dHorizontal` and `-Vertical` have an inverse attribute `ToAlignmentCurve FOR IfcAlignmentCurve` 
- when generated, both receive an inverse attribute `SET<IfcAlignmentCurve> ToAlignmentCurve`
- `IfcAlignmentCurve::linkInverse` links itself to the corresponding inverse attribute of `-Horizontal` and `-Vertical`

Build sequence:
- build `OpenInfraPlatform.ExpressBindingGenerator`
- build `Generate.schema_UT_v3`
- see the generated files (`UnitTests\Generator\schema_UT_v3\Generated`)

### CODE REVIEW

File: ExpressBindingGenerator/src/Generator/GeneratorOIP.cpp [hunk 1]
```diff
@@ -4102,13 +4124,11 @@ void GeneratorOIP::generateEntitySourceFileREFACTORED(Schema & schema, const Ent
...
 		linebreak(out);
 
-		auto attributes = schema.getAllEntityAttributes(entity);
-
 		// Copy Assignment Operator
 		writeDoxyComment(out, "Assigns the content of \\c other to \\c this.");
 		writeLine(out, entity.getName() + "& " + entity.getName() + "::operator=(const " + entity.getName() + "& other) {");
 		writeLine(out, "this->m_id = other.m_id;");
-		for (auto& attr : attributes) {
+		for (auto& attr : schema.getAllEntityAttributes(entity, true)) {
```
Comment (hechth) [inline ExpressBindingGenerator/src/Generator/GeneratorOIP.cpp]: Should be again pulled out of the loop because this is probably fairly slow

Comment (pjanck) [reply ExpressBindingGenerator/src/Generator/GeneratorOIP.cpp]: Resolved 3a074828

File: ExpressBindingGenerator/src/Generator/GeneratorOIP.cpp [hunk 2]
```diff
@@ -3534,7 +3556,7 @@ void GeneratorOIP::generateEntityHeaderFileREFACTORED(Schema & schema, Entity &
 	
 	//writeLine(out, "typedef " + entity.getName() + " UnderlyingType;");
 	
-	auto initClassAsDefault = [&out, &entity, &schema]() {
```
Comment (hechth) [inline ExpressBindingGenerator/src/Generator/GeneratorOIP.cpp:3571]: Where did this go?

Comment (pjanck) [reply ExpressBindingGenerator/src/Generator/GeneratorOIP.cpp:3571]: I removed the function definition and just left the body. (The function was called once, directly after its definition).

**Comment by pjanck** [comment]
Superseded by #274 .

**Pull request closed without merging**